### ISSUE TITLE
quote.getCalculatedTotal() does not include shipping price

### ISSUE THREAD

**Issue by jstuardo** [magento/magento2#4456]
I have implemented a payment method whose method renderer has this function:

```
                getQuoteAmount: function() {
                    return quote.getCalculatedTotal() + '.00';
                },
```

The problem is that quote.getCalculatedTotal()  does not include shipping price. Is this a bug or do I need to call other function? I tried searching for other function but without success.

Any help please?
Thanks
Jaime

**Comment by cpartica** [comment]
quote.getCalculatedTotal does not include shipping ,that's true if you look at Magento/Checkout/view/frontend/web/js/model/quote.js

```
            setTotals: function(totalsData) {
                if (_.isObject(totalsData) && _.isObject(totalsData.extension_attributes)) {
                    _.each(totalsData.extension_attributes, function(element, index) {
                        totalsData[index] = element;
                    });
                }
                totals(totalsData);
                this.setCollectedTotals('subtotal_with_discount', parseFloat(totalsData.subtotal_with_discount));
            },
```

it only has one key: the subtotal with discount that's without shipping but quote.getCalculatedTotal() would collect all keys added to the collectedTotals() observable object

We are investigating this if the expected behavior or is it a bug

**Comment by veloraven** [comment]
@jstuardo is this issue still actual?
If it is not, please close the issue.
If it is, please provide the used Magento version and description according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result.

**Comment by andimov** [comment]
According to [contributor guide](http://devdocs.magento.com/guides/v2.1/contributor-guide/contributing.html), tickets without response for two weeks should be closed.
Please feel free to reopen if it's needed.

**Comment by delgadito13** [comment]
Hi, 

We still get this issue for a project we are doing now and we see that indeed the totalsData object does not include the shipping cost at all at checkout. Only after a reload of the page it seems to be picking up the data, but that action results also in loss of information of some form details, as the billing address at doing so.
Please have another look at this one. Please contact me in case you need any more details.

Not sure if and how I can open this ticket.

**Comment by veloraven** [comment]
@delgadito13 please create a new issue formatting it according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result.
Please, also identify which version of Magento you are running.

**Comment by SerhiyShkolyarenko** [comment]
Dynamically updated data is stored in quote.totals. See details in #6796.
